File: sharp_brain/sharp_brain/Neural/WorkingNeuron.cs
using SharpBrain.Activations;
using System.Collections.Generic;

namespace SharpBrain.Neural
{
    public abstract class WorkingNeuron : Neuron
    {
        protected readonly Activation activation;

        public List<Connection> Connections
        {
            get; set;
        }

        protected WorkingNeuron(int layerIndex, int index, Activation activation)
            : base(layerIndex, index)
        {
            this.activation = activation;
        }

        public override double Activate()
        {
            double s = 0.0;
            foreach (var con in Connections)
            {
                s += con.Activate();
            }
            netInput = s;
            Value = activation.Activate(s);
            return Value;
        }

        public void ApplyDelta()
        {
            foreach (var con in Connections)
            {
                var deltaWeight = ETA * delta * con.NeuronValue;
                con.AdjustWeight(deltaWeight);
            }
        }
    }
}

File: kotlin_brians/src/main/kotlin/neurals/WorkingNeuron.kt
package neurals

import activations.Activation

abstract class WorkingNeuron(layerIndex: Int, index: Int, protected val activation: Activation)
    :Neuron(layerIndex, index) {
    var delta: Double = 0.0
    var netInput: Double = 0.0

    private val eta: Double = 1.0

    lateinit var connections: List<Connection>

    override fun activate(): Double {
        var s = 0.0
        for (con in connections) {
            s += con.activate()
        }
        netInput = s
        value = activation.activate(s)
        return value
    }

    fun applyDelta() {
        for(c in connections) {
            val adjustment: Double = eta * delta * c.inputNeuron.value
            c.adjustWeight(adjustment)
        }
    }
}